### ISSUE TITLE
Gephi hangs when you switch from Data Laboratory to Overview

### ISSUE THREAD

**Issue by richgor** [gephi/gephi#1253]
Running OSX 10.9.5, I have had this same experience on multiple data sets.  Using the simplest possible example:
* I open the Les Miserables.gexf sample data file
* The import wizard opens and I click "OK"
* The import appears to be successful - I see the graph visualization
* I click over into the Data Laboratory - nodes and edges tables appear to be in order
* I click back to Overview and Gephi hangs.  
* The only way to get out is to Force Quit.

I had the same experience with larger datasets not originally provided as samples by Gephi.

For what it's worth, I also tried to register for your support forums but no matter what machine or login (or network) I use, I am blocked because it thinks I am a spammer.  Which I am not.

**Comment by richgor** [comment]
I have one additional piece of information that may (or may not) help you diagnose the problem.  Today, when I had this problem, was the second day I had tried to work with the new Gephi 0.9.  Last week (Dec. 21), I worked quite a bit with Gephi and didn't have this problem at all.  As far as I know, the only thing that changed between Dec. 21 and Dec. 27 was that I installed several updates in response to the prompt in the lower right of my screen (see attached screenshot below).   

If I use the 2nd Mac computer where I had installed Gephi on Dec. 21 (pre-updates), I DO NOT have the "hang" problem.  So I am wondering if the problem was caused by one of the updates between Dec. 21 and Dec. 27.

I am *not* going to install any of the updates on the 2nd Mac for now.  If there is a way to revert to the pre-update version of the software, please let me know.

![updates-prompt-dec27](https://cloud.githubusercontent.com/assets/1087501/12012478/d5219f92-acbb-11e5-88c6-4ef0f620ef8f.png)

**Comment by richgor** [comment]
Today I have been using Gephi 0.9 on a workplace Mac running OS X Yosemite (10.10.5).  I have *not* installed the 5 updates that are shown in the lower right of my Gephi window.

Without those updates installed, Gephi is working fine - the "hang" problem is not occurring.

To recap:
* On my secondary home Mac running OS X El Capitan, and *no* updates installed, Gephi is working fine.
* On a workplace Mac running OS X Yosemite, and *no* updates installed, Gephi is working fine.
* On my primary Mac running OS X Mavericks, and *with* updates installed, Gephi hangs.
* According to this issue (https://github.com/gephi/gephi/issues/1250), user **stared** is having the "hang" problem on OS X Mountain Lion and on OS X El Capitan; and user **buethastum** is having the same problem on OS X Mavericks.  I do not know if these users have installed the newest updates, but I just posted a query to that effect.
* According to this issue (https://github.com/gephi/gephi/issues/1241), multiple users on different Mac OS's are having the "hang" problem. I do not know if these users have installed the newest updates, but I just posted a query to that effect.

Anyway, pending hearing back from the users in the other issue threads, my experience suggests that the "hang" problem is related to the updates made available after Dec. 20.  If you have NOT yet installed those updates, I would recommend that you not do so.  And if someone knows how to revert to the pre-updated version, I would love to know how.  (I have tried uninstalling and reinstalling from the original .dmg but that didn't work.)

**Comment by dkbohl** [comment]
Thanks @richgor, I have also installed the updates.

**Comment by richgor** [comment]
@dkbohl - Evidence growing that something in those updates is causing the problem we are having.   Wish I knew how to revert to the pre-updated version.  Hope @mbastian or someone else on the coding team can diagnose and fix swiftly!

**Comment by eduramiba** [comment]
Hi, this seems to be related to JOGL, and by looking at the updates we did, the only change to visualization was to hide the fps by default, so I can't find a direct cause of the issue by just taking the updates into account.

There is surely a JOGL problem that happens to some OS X users, as seen in #1241, and it happened before the updates.

You can remove them (and all preferences) by clearing your user files directory. See https://github.com/gephi/gephi/wiki/Troubleshooting

**Comment by richgor** [comment]
Thanks @eduramiba.  You certainly understand this stuff better than I do, and sorry if my suggestions are off-target.  But the fact remains that the only Mac computer where I'm having the "hang" problem is the one where I did install the updates -- and I have no problems on the Macs where I didn't install the updates.  

Thanks for the pointer for how to clear my user files.  I will do that.  One clarification question: If I clear my user files and then download the current .dmg from Gephi.org, installing from that .dmg will *not* include the updates?

**Comment by eduramiba** [comment]
No problem :)

Just trying to understand how the issue may be linked to the updates. The fps change might be making it easier for the issue to trigger? Not sure...

When you delete the directory, close Gephi and start it again. Then it should be like you just had installed it. No need to reinstall the dmg (does not include updates either).

Can you try to not install updates in the first Mac and see if you still can reproduce the problem? It's probably caused by different hardware on each of your Macs so you should be getting the same problem without the updates on the first Mac.

**Comment by richgor** [comment]
@eduramiba I deleted the user files directory, relaunched Gephi and now it is working fine -- no "hang" when I switch from the Data Laboratory to the Overview.  

I do see a little red icon in the lower right of my screen that, when I mouse over it, says "Unexpected Exception" (PNG attached).  Actually, let me update that - the icon has now disappeared (though the PNG shows what I saw).  

Anyway, it seems like I can work with Gephi just fine as long as I don't install those updates.  In case it's helpful, I will provide information below on the specs for the three different Macs involved (one laptop, two desktops): 

(1) my primary laptop - a MacBook Pro running OS X 10.9.5 (Mavericks) with these specs:
  Model Identifier:	MacBookPro11,3
  Processor Name:	Intel Core i7
  Processor Speed:	2.6 GHz
  Number of Processors:	1
  Total Number of Cores:	4
  L2 Cache (per Core):	256 KB
  L3 Cache:	6 MB
  Memory:	16 GB

(2) Work desktop: coming soon

(3) Home desktop: coming soon

![gephi-unexpected-exception](https://cloud.githubusercontent.com/assets/1087501/12025734/539c103a-ad77-11e5-9169-373be6d9d012.png)

**Comment by richgor** [comment]
Update @eduramiba: I was able to update the Data Laboratory module (*without*) the Appearance API, the Visualization Module and the Data Laboratory Plugin.  With these updates, the import problem (weights not being imported) was solved -- AND I do not have the "hang" problem when switching from Data Laboratory to Overview.  

This suggests that whatever is causing the "hang" problem is somehow related to either the Appearance API or the Visualization Module.  Hope you can find a solution!

**Comment by richgor** [comment]
@mbastian @eduramiba Any updates on fixing this "hang" problem when switching from Data Laboratory to Overview?

**Comment by mbastian** [comment]
The bug isn't likely related to updates because the that code wasn't changed. It's just that with most graphical JOGL bugs there's some randomness attached to it. That's why I can't reproduce this bug on the couple Macs I personally I have access to. Sometimes it would work and sometimes it wouldn't. The fact it worked without the update is just pure luck in my opinion.

**Comment by richgor** [comment]
Thanks for the reply @mbastian.  You obviously know way more about the technical back-end than I do, but it seems to me that if the behavior is identical on three different Macs (two desktops, one laptop) running three different versions of the Mac OS, it's something more than "randomness."  Whatever the cause, I hope you can find a solution.  I am now locked into using Gephi for a class I am teaching that starts tomorrow - I'm praying that the students will be able to do their assignments without having this problem themselves.  If you have any suggestions on how to avoid the problem or how to diagnose and fix it, they are welcome.  And if there's anything else I can send you (log files or whatever) that would be helpful, let me know.

**Comment by mbastian** [comment]
I understand your case but I'm still convinced this has nothing to do with the updates. Proof is that the problem was reported way before we created those updates. I also want to stress that we don't want users to be afraid of installing updates. Rather the opposite in fact. An update might break something sometimes but that's not a reason for not installing them. A subsequent update would then bring things back to normal. I understand your frustration as your class starts soon but I hope to fix this asap. I plan to work on it this week on a fast pace. Unfortunately, there isn't a workaround at this point. Thanks for your understanding.

**Comment by Nyzl** [comment]
just chipping in here, i was having the same issue and was driving me up the wall, also on OS X 10.11
Even doing a fresh install with no updates didn't help.
What did help was deleting the user files. I then installed the updates and everything is working again. Sorry it's not working for everyone
Happy to try and replicate, and if it happens again i'll share what i was doing.

**Comment by computermacgyver** [comment]
@richgor you asked in another thread how to uninstall updates to Gephi. To do so, simply delete ~/Library/Application Support/gephi/

**Comment by richgor** [comment]
Thanks @computermacgyver.  Got that message earlier and have done it several times.  Seems to be the quickest way to get rid of the "hang" problem that I and others have had.

**Comment by Nyzl** [comment]
@mbastian i've managed to replicate the issue without plugins installed. It seems to occur when switching between the Overview, Data Laboratory and Preview tabs when Gephi is maximised. (there seems to be some behaviour where Gephi maximises itself the first time you click on one of those tabs). If you manually resize the window the hang error does not occur.

I should add that I am using dual screens (Mac book and monitor)

**Comment by mbastian** [comment]
This issue is duplicate of #1241

**Comment by rhoconlinux** [comment]
Hello guys, I´m still having this issue in Monterrey. Similar diagnostic. 
Any clue on how to solve it, would be great. I´ve already deleted my user settings about 7 times, it´s the only trick that let me relaunch the program. 
Cheers!

**Comment by ryac** [comment]
Hi, I am having the same issue on MacOS 12.4 and Gephi 0.9.6. The application will hang when I switch between Overview and Data Laboratory only after a project is loaded. Sometimes I will be able to see the Data Laboratory screen, but more often it will hang. Could it be the Java version or memory allocation? More details below:

Product Version: Gephi 0.9.6 202206221744
Java: 11.0.15; OpenJDK 64-Bit Server VM 11.0.15+10
Runtime: OpenJDK Runtime Environment 11.0.15+10
System: Mac OS X version 12.4 running on x86_64; UTF-8; en_GB (gephi)
